common/src/main/java/com/microsoft/identity/common/internal/broker/InactiveBrokerClient.java

```diff
@@ -0,0 +1,177 @@
+ **/
+public class InactiveBrokerClient {
+    private static final String TAG = InactiveBrokerClient.class.getSimpleName();
+
+    private static final String BROKER_ACCOUNT_SERVICE_INTENT_FILTER = "com.microsoft.workaccount.BrokerAccount";
+    private static final String BROKER_ACCOUNT_SERVICE_CLASS_NAME = "com.microsoft.aad.adal.BrokerAccountService";
+
+    private Context mContext;
+    private String mCallingPackageName;
+    private BrokerAccountServiceConnection mBrokerAccountServiceConnection;
+    private Intent mBrokerAccountServiceIntent;
+    private Boolean mBound = false;
+
+    // Must be called from BG thread.
+    public static String getSerializedSymmetricKeyFromInactiveBroker(@NonNull Context context,
+                                                                     @NonNull final String activeBrokerPackageName) {
+        final String methodName = ":getSerializedSymmetricKeyFromInactiveBroker";
+
+        final InactiveBrokerClient client = new InactiveBrokerClient(context, activeBrokerPackageName);
+        try {
+            final BrokerAccountServiceFuture brokerAccountServiceFuture = client.connect();
+            final IBrokerAccountService service = brokerAccountServiceFuture.get();
+
+            final Bundle resultBundle = service.getInactiveBrokerKey();
+
+            // This means that the inactive broker doesn't support getInactiveBrokerKey().
+            if (resultBundle == null){
+                Logger.verbose(TAG + methodName, "resultBundle is null");
+                return null;
+            }
+            else if (resultBundle.getString(AuthenticationConstants.OAuth2.ERROR) != null) {
+                // NOTE: BrokerAccountService's createErrorBundle is still using OAuth2.ERROR. so i'm keeping the pattern here.
+                Logger.error(
+                        TAG + methodName,
+                        "Receiving error result from inactive broker: "
+                                + resultBundle.getString(AuthenticationConstants.OAuth2.ERROR_DESCRIPTION),
+                        null);
+                return null;
+            }
+
+            return resultBundle.getString(AuthenticationConstants.Broker.BROKER_KEYSTORE_SYMMETRIC_KEY);
+
+        } catch (final BaseException | InterruptedException | ExecutionException | RemoteException e) {
+            Logger.error(
+                    TAG + methodName,
+                    "Exception is thrown when trying to get key from inactive broker:"
+                            + e.getMessage(),
+                    e);
+        } finally {
+            client.disconnect();
+        }
+
+        return null;
+    }
+
+    /**
+     * Constructor for the BrokerAccountServiceClient
+     *
+     * @param context
+     * @param callingPackageName name of the calling package.
+     */
+    private InactiveBrokerClient(@NonNull final Context context,
+                                 @NonNull final String callingPackageName) {
```

Comment:
`callingPackageName` has been renamed to `activeBrokerPackageName` in other places. May be we can also make that change here.